test/unit/unpublishing_test.rb

```diff
@@ -145,6 +145,29 @@ class UnpublishingTest < ActiveSupport::TestCase
     assert_equal original_path, unpublishing.document_path
   end
 
+  test '#document_url returns the URL for the unpublished edition' do
+    edition = create(:detailed_guide, :draft)
+    original_url = Whitehall.url_maker.public_document_url(edition)
+    unpublishing = create(:unpublishing, edition: edition,
+                          unpublishing_reason: UnpublishingReason::PublishedInError)
+
+    assert_equal original_url, unpublishing.document_url
+  end
+
+  test '#document_url returns the URL for a deleted unpublished edition' do
+    edition = create(:detailed_guide)
+    original_url = Whitehall.url_maker.public_document_url(edition)
+    unpublishing = create(:unpublishing, edition: edition,
+                          unpublishing_reason: UnpublishingReason::PublishedInError)
+
+    EditionDeleter.new(edition).perform!
+    # The default scope on Edition stops deleted editions being found when an
+    # unpublishing is loaded. To trigger the bug we need to reload.
+    unpublishing.reload
+
+    assert_equal original_url, unpublishing.document_url
+  end
+
```

Comment:
I've deleted the "path" version of the test for a "deleted unpublished edition" in https://github.com/alphagov/whitehall/pull/3840, although it's not yet merged. If this is merged after that PR, it would be good to delete the "URL" version of the same test before merging this PR.

Reply:
@floehopper. I've merged this PR so I'll leave it to you to decide whether to delete this test as part of #3840.

Reply:
@chrisroos I have deleted this test in 88d1a0fd8d262e738ebb41ff90c05fe9e67ab7f6.